collector/pkg/config/config.go

```diff
@@ -48,6 +48,16 @@ func (c *configuration) Init() error {
 	c.SetDefault("commands.metrics_info_args", "--info --json")
 	c.SetDefault("commands.metrics_smart_args", "--xall --json")
 
+	// Bind environment variables with COLLECTOR_ prefix
+	c.BindEnv("host.id", "COLLECTOR_HOST_ID")
+	c.BindEnv("log.level", "COLLECTOR_LOG_LEVEL")
+	c.BindEnv("log.file", "COLLECTOR_LOG_FILE")
+	c.BindEnv("api.endpoint", "COLLECTOR_API_ENDPOINT")
+	c.BindEnv("commands.metrics_smartctl_bin", "COLLECTOR_COMMANDS_METRICS_SMARTCTL_BIN")
+	c.BindEnv("commands.metrics_scan_args", "COLLECTOR_COMMANDS_METRICS_SCAN_ARGS")
+	c.BindEnv("commands.metrics_info_args", "COLLECTOR_COMMANDS_METRICS_INFO_ARGS") 
+	c.BindEnv("commands.metrics_smart_args", "COLLECTOR_COMMANDS_METRICS_SMART_ARGS")
+	
```

Comment:
```suggestion
	// Bind environment variables with COLLECTOR_ prefix
	c.SetEnvPrefix("COLLECTOR")
```

I think that should work instead. Can you test and confirm @datenzar ?

Reply:
Hi @AnalogJ 
sorry for my late reply, but I haven't had the chance to look into this issue for a while.

From what I understood by reading in the Viper documentation, the prefix alone isn't sufficient. 

The following two lines however should do the job
```go
c.SetEnvPrefix("COLLECTOR")
c.AutomaticEnv()
```

I tried the above lines, but the values weren't picked up. I'm not sure if I misunderstood the concept but following call (on Linux) should lead to an error due to an arbitrary non-existing command being used:
`COLLECTOR_COMMANDS_METRICS_SMARTCTL_BIN=non_existing_cmd go run collector/cmd/collector-metrics/collector-metrics.go run --config collector.yaml`

Yet the application won't pick up the value from env. BTW: It works perfectly, when removing the hierarchy (-> replace all keys `commands.metrics_smartctl_bin` with `commands_metrics_smartctl_bin`).

Reply:
NVM, I was able to fix it with a code piece I stumbled on your webapp backend. I was able to test it with success 🥳